pkg/streamingpromql/operators/functions/range_vectors.go

```diff
@@ -67,17 +67,23 @@ func presentOverTime(step *types.RangeVectorStepData, _ float64, _ types.EmitAnn
 }
 
 var MaxOverTime = FunctionOverRangeVectorDefinition{
-	SeriesMetadataFunction: DropSeriesName,
-	StepFunc:               maxOverTime,
+	SeriesMetadataFunction:         DropSeriesName,
+	StepFunc:                       maxOverTime,
+	NeedsSeriesNamesForAnnotations: true,
 }
 
-func maxOverTime(step *types.RangeVectorStepData, _ float64, _ types.EmitAnnotationFunc) (float64, bool, *histogram.FloatHistogram, error) {
+func maxOverTime(step *types.RangeVectorStepData, _ float64, emitAnnotation types.EmitAnnotationFunc) (float64, bool, *histogram.FloatHistogram, error) {
 	head, tail := step.Floats.UnsafePoints()
 
 	if len(head) == 0 && len(tail) == 0 {
 		return 0, false, nil, nil
 	}
 
+	hHead, hTail := step.Histograms.UnsafePoints()
```

Comment:
This could be simplified to use `step.Histograms.Any()`.

(same applies below and to the change for `deriv` in https://github.com/grafana/mimir/pull/10383/files)

Reply:
Much nicer. Done!